### PR TITLE
add optional 'ca' tls directive, closes #1689

### LINKED ISSUE #1689: TLS CAUrl Option
Currently the ACME CA URL is configured via the `-ca` command line argument. This means changing the value involves restarting the process and that all domains must use the same ACME CA.  I think this option would be better suited for the `tls` directive, e.g.:

```
staging.mydomain.com {
  tls {
    ca https://acme-staging.api.letsencrypt.org/directory
  }
  ...
}

other.mydomain.com {
  tls {
    ca https://acme-v01.api.letsencrypt.org/directory
  }
  ...
}
```

I think I can whip up a PR for this idea if it has value.

### PR DESCRIPTION
**Pull request by zikes**
### 1. What does this change do, exactly?
This change adds an optional `ca` option to the `tls` directive, which lets you specify an ACME endpoint within the Caddyfile.

### 2. Please link to the relevant issues.

#1689 

### 3. Which documentation changes (if any) need to be made because of this PR?

The new option should be added at https://caddyserver.com/docs/tls

### 4. Checklist

- [x] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I am willing to help maintain this change if there are issues with it later

### CODE REVIEW

File: caddytls/setup_test.go [hunk 1]
```diff
@@ -277,6 +277,25 @@ func TestSetupParseWithClientAuth(t *testing.T) {
 	}
 }
 
+func TestSetupParseWithCAUrl(t *testing.T) {
+	testUrl := "https://acme-staging.api.letsencrypt.org/directory"
```
Comment (elcore) [inline caddytls/setup_test.go]: Rename 'testUrl' to 'testURL'

Comment (zikes) [inline caddytls/setup_test.go]: @elcore Updated and squashed

**Comment by mholt** [comment]
Welcome @zikes! Thanks for your contribution. I'll give this a second review before long. :)

File: caddytls/setup.go [hunk 2]
```diff
@@ -66,6 +66,12 @@ func setupTLS(c *caddy.Controller) error {
 		for c.NextBlock() {
 			hadBlock = true
 			switch c.Val() {
+			case "ca":
+				arg := c.RemainingArgs()
```
Comment (mholt) [inline caddytls/setup.go]: We should use only the first argument; if there are more or less than 1 argument, it should be an error.

Comment (elcore) [inline caddytls/setup.go]: Sounds good to me :)

Comment (zikes) [inline caddytls/setup.go]: I've replaced the check with `len(args) != 1` similar to how `dns` and `storage` check their args, and added test cases for too many/too few args.

**Comment by mholt** [comment]
Thank you @zikes! If you have a chance, submit a PR to https://github.com/caddyserver/website to add the docs for the new `ca` option. :)

**Pull request merged**